pandas/tests/indexing/test_categorical.py

```diff
@@ -806,3 +806,13 @@ def test_loc_with_non_string_categories(self, idx_values, ordered):
         result.loc[sl, "A"] = ["qux", "qux2"]
         expected = DataFrame({"A": ["qux", "qux2", "baz"]}, index=cat_idx)
         tm.assert_frame_equal(result, expected)
+
+    def test_align_keeps_categorical_index(self):
+        # GH-28397
+        df_1, df_2 = self.df.copy(), self.df.copy()
```

Comment:
Can you use two different dataframes? 
(as otherwise it might take a shortcut of not needing any alignment when the indexes are identical)

Reply:
That's a good point, in implementing a test for that I stumbled on the following when aligning

If the indices have the same categories: the resulting index has `dtype=category`
if the indices have different categories: the resulting index has `dtype=object`

Is that the desired behaviour? If so, I'll implement a test with different dataframes but with the same categories. If not, I'll have to dig to see where this needs to be fixed.

Reply:
yes this is correct behavior